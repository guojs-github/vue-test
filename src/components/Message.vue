<template>
	<div id="message-list">
		<MessageList :messages="messages"/>
	</div>
</template>

<script>
import MessageList from './MessageList'

export default {
	name: 'Message',
	data: () => ({ messages: ['Hey John', 'Howdy Paco'] }),
	components: {
		MessageList
	}
}
</script>