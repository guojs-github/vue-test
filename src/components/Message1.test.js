import { shallowMount } from "@vue/test-utils";
import Message from "./Message";

describe("Message1.test.js", () => {
	let cmp;

	beforeEach(() => {
		cmp = shallowMount(Message, {
			// Create a shallow instance of the component
			data: () => ({ messages: ['Cat'] })
		});
	});

	it('equals messages to ["Cat"]', () => { // 检验组件数据是否符合要求
		expect(cmp.vm.messages).toEqual(["Cat"]);
	});

	it("has the expected html structure", () => { // 检验html显示结构是否符合要求
		expect(cmp.element).toMatchSnapshot();
	});	
});