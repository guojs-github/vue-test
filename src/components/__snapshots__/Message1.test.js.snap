// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message1.test.js has the expected html structure 1`] = `
<div
  id="message-list"
>
  <messagelist-stub
    messages="Cat"
  />
</div>
`;
